src/devices/cpu/tms32025/tms32025.cpp

```diff
@@ -184,7 +184,16 @@ Table 3-2.  TMS32025/26 Memory Blocks
 #define IND     m_AR[ARP]                       /* address used in indirect memory access operations */
 
 
+/*
+   Processor can be operated in one of two modes based on Pin 1 (MP/MC)
+   MP/MC = 1 (Microprocessor Mode)
+   MP/MC = 0 (Microcomputer Mode)
+   in 'Microcomputer' mode the 4K Word internal ROM is used (TMS320C25)
+
+   using a different device type for each as MAME doesn't appear to offer an easy way to handle this otherwise?
```

Comment:
That's not a very nice workaround. What methods did you try? If there really is no other way, can't you pretend this internal rom is external?

Reply:
I'd rather not pretend the internal ROM was external, IMHO that's an uglier workaround.

I could see no way to have a different internal map depending on the configuration however.  It's fixed configuration afaik (you can't change the pin state after boot) so this seemed the most logical way.

Reply:
the only other workaround seem to actually just be exploiting bugs in MAME with regards external ROMs mapping over internal ROMs and I'd *really* rather not do that

Reply:
so yes, for the time being I think this is the cleanest solution by far

Reply:
unless I install ROM in the device, but that seems to be frowned upon these days?  would resolve your issue tho.

Reply:
well I've done that, not sure I'm actually happier with that solution, but it still works without breaking everything elsewhere or relying on weird side-effects.